tests/datasource/fluent/test_batch_request.py

```diff
@@ -113,7 +124,85 @@ def test_batch_request_config_serialization_round_trips(
         '"sort_ascending": true, '
         '"method_name": "partition_on_column_value"'
         "}, "
-        '"batching_regex": "data_(?P<year>\\\\d{4})-(?P<month>\\\\d{2}).csv", '
         f'"batch_slice": {batch_slice_json}'
         "}"
     )
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "partitioner,TypedBatchRequest",
+    [
+        pytest.param(
+            PartitionerYearAndMonthAndDay(
+                column_name="foo",
+                sort_ascending=False,
+            ),
+            BatchRequest[Partitioner],
+            id="Sql Daily",
+        ),
+        pytest.param(
+            PartitionerYearAndMonth(
+                column_name="foo",
+                sort_ascending=False,
+            ),
+            BatchRequest[Partitioner],
+            id="Sql Monthly",
+        ),
+        pytest.param(
+            PartitionerYear(
+                column_name="foo",
+                sort_ascending=False,
+            ),
+            BatchRequest[Partitioner],
+            id="Sql Yearly",
+        ),
+        pytest.param(
+            PartitionerDaily(
+                regex=re.compile(r"data_(?P<year>\d{4})-(?P<month>\d{2})-(?P<day>\d{2}).csv"),
+                sort_ascending=False,
+            ),
+            BatchRequest[RegexPartitioner],
+            id="Regex Daily",
+        ),
+        pytest.param(
+            PartitionerMonthly(
+                regex=re.compile(r"data_(?P<year>\d{4})-(?P<month>\d{2}).csv"),
+                sort_ascending=False,
+            ),
+            BatchRequest[RegexPartitioner],
+            id="Regex Monthly",
+        ),
+        pytest.param(
+            PartitionerYearly(
+                regex=re.compile(r"data_(?P<year>\d{4}).csv"),
+                sort_ascending=False,
+            ),
+            BatchRequest[RegexPartitioner],
+            id="Regex Yearly",
+        ),
+        pytest.param(
+            None,
+            BatchRequest[None],
+            id="None",
+        ),
+    ],
+)
+def test_batch_request_config_partitioner_round_trip_serialization(
```

Comment:
Having this functionality under test is great. But is this testing real-world use cases for deserialization? Will we always have the equivalent of `TypedBatchRequest` at deserialization time? My hunch is "no", but I'm guessing you've thought ahead on this.

Also obligatory nits:
* why not split this into 2 tests: one for dicts and one for json
* the constants make me think those strings will be important/reused, but they aren't.

Reply:
> is this testing real-world use cases for deserialization?
right, this is what i brought up at standup today. I wrote this test because i wanted to know if Pydantic needs the type to deserialize -- it doesn't, but mypy does. We're avoiding this right now by deserializing batch configs by magic in checkpoints. In the future, i would suggest the Asset/BatchDefinition should always be responsible for retrieving batch configs, and they will always have the correct type context.

Reply:
re: those nits - i copied the structure of the test above it 🙃 i'll update to address.

Reply:
[001190e](https://github.com/great-expectations/great_expectations/pull/9792/commits/001190e4ecf3feeda131ba19efd2e84f60d77f42)